File: 백준/Silver/10655. 마라톤 1/마라톤 1.py
import sys


n = int(input())
check_points = [list(map(int, input().split())) for _ in range(n)]

def get_distance(x1, y1, x2, y2):
    return abs(x1-x2) + abs(y1-y2)

total = 0
for i in range(n-1):
    x1, y1 = check_points[i]
    x2, y2 = check_points[i+1]
    total += get_distance(x1, y1, x2, y2)
# print(total)

min_distance = float('inf')
for i in range(1, n-1): # 제외할 체크포인트 지점
    px, py = check_points[i]
    prev_x, prev_y = check_points[i-1]
    next_x, next_y = check_points[i+1]

    temp = total - get_distance(px, py, prev_x, prev_y) - get_distance(next_x, next_y, px, py)
    temp += get_distance(next_x, next_y, prev_x, prev_y)
    min_distance = min(min_distance, temp)
print(min_distance)


# i+2 를 건너뛰면 => i+2 ~ i+1, i+3 ~ i+2 만큼 빼주고
#                 i+3 ~ i+1 만큼 더해주고

File: 백준/Silver/10655. 마라톤 1/마라톤 1.java
import java.io.*;
import java.util.*;

public class Main {

    static class Point {
        int x, y;
        Point(int x, int y) {
            this.x = x;
            this.y = y;
        }
    }

    static int getDistance(Point a, Point b) {
        return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        Point[] checkPoints = new Point[n];

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            checkPoints[i] = new Point(x, y); // 여기
        }

        int total = 0;
        for (int i = 0; i < n-1; i ++) {
            total += getDistance(checkPoints[i], checkPoints[i+1]);
        }

        int minDistance = Integer.MAX_VALUE;

        for (int i = 1; i < n-1; i++) {
            int original = getDistance(checkPoints[i - 1], checkPoints[i]) +
                    getDistance(checkPoints[i], checkPoints[i + 1]);
            int shortcut = getDistance(checkPoints[i - 1], checkPoints[i + 1]);
            int newTotal = total - original + shortcut;

            minDistance = Math.min(minDistance, newTotal);
        }

        System.out.println(minDistance);

    }
}